simpleclient_common/src/main/java/io/prometheus/client/exporter/common/TextFormat.java

```diff
@@ -2,37 +2,108 @@
 
 import java.io.IOException;
 import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 
 import io.prometheus.client.Collector;
 
 public class TextFormat {
   /**
-   * Content-type for text version 0.0.4.
+   * Content-type for Prometheus text version 0.0.4.
    */
   public final static String CONTENT_TYPE_004 = "text/plain; version=0.0.4; charset=utf-8";
 
+  /**
+   * Content-type for Openmetrics text version 1.0.0.
+   */
+  public final static String CONTENT_TYPE_OPENMETRICS_100 = "application/openmetrics-text; version=1.0.0; charset=utf-8";
+
+  /**
+   * Return the content type that should be used for a given Accept HTTP header.
+   */
+  public static String chooseContentType(String acceptHeader) {
+    if (acceptHeader == null) {
+      return CONTENT_TYPE_004;
+    }
+
+    for (String accepts : acceptHeader.split(",")) {
+      if ("application/openmetrics-text".equals(accepts.split(";")[0].trim())) {
```

Comment:
nit: Worth making this a const? It's used both in the content-type + version string const as well as here when detecting the accept type.

Reply:
I don't think it'd aid readability to have a constant based on other constants, this is a well known string - I find it easier to have the raw text rather than digging through layers of indirection to see what this code is actually doing.